app/models/post.rb

```diff
@@ -0,0 +1,29 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :text, presence: true
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  before_validation :set_timestamps
+
+  after_save :update_posts_counter
+
+  def set_timestamps
+    self.created_at ||= Time.current
+    self.updated_at = Time.current
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ Required ] I think it would be nice if you ensure that all custom methods in your models are also tested. To comply with the project requirements please add unit tests for the `recent_comments` and `update_posts_counter` methods in the `Post` model.

## Example code suggestion for recent_comments method
```ruby
  it 'should return a users last 5 comments' do
    user = User.create( name: 'Alex', bio: 'Graphic designer',
                        photo: 'http://hello.com/org.png', posts_counter: 0)
    post = Post.create(title: 'A new dawn', text: 'Software Developer', author_id: user.id, 
                       comments_counter: 0,  likes_counter: 0)

    10.times.collect do
      Comment.create(text: 'Hello, world', author_id: user.id, post_id: @post.id)
    end
    expect(@post.recent_comments.length).to eq(5)
  end
```

## Project requirements
![Screenshot from 2023-11-18 13-10-59](https://github.com/paulshaun-kdtk/blog-posts/assets/16392046/3884496b-33ca-40bd-b132-dbf684fa13c0)